adoc/admin-disaster-recovery-restore-master-node.adoc

```diff
@@ -0,0 +1,108 @@
+
+== Recover Master Nodes - etcd cluster and data
+Ensure cluster version for backup/restore should be the same. Cross-version restoration in any domain is likely to  encounter data/API compatibility issues. 
+
+=== Prerequisite
+Only need to restore database on one of the master node (`master-0`) to regain control-plane access. 
+
+The remain master node will sync to the restored database after joined to cluster. 
+
+Do the following on `master-0`.
+
+1. Install `etcdctl`.
+```bash
+sudo zypper install etcdctl
+```
+2. Have a access of etcd snapshot from backup device.
+
+=== Procedure
+1. Stop etcd on all master nodes.
++
+```bash
+mv /etc/kubernetes/manifests/etcd.yaml /tmp/
+```
+You can check etcd container does not exist with `crictl ps | grep etcd`
+2. Purge etcd data on all master nodes.
++
+```
+sudo rm -rf /var/lib/etcd
+```
+3. Remote to `master-0`.
+4. Restore etcd data.
++
+Replace <SNAPSHOT> with path to etcd snapshot, for example: /share/backup/etcd_snapshot/etcd-snapshot-2019-11-08_05:19:20_GMT.db
++
+Replace <NODE_NAME> with `master-0` cluster node name, for example: skuba-master-1
++
+Replace <NODE_IP> with `master-0` cluster node IP address.
++
+HINT: the <NODE_NAME> and <NODE_IP> must match the correct node information and should exist after `--initial-cluster` in /etc/kubernetes/manifest/etcd.yaml 
++
+```bash
+SNAPSHOT="<SNAPSHOT>"
+NODE_NAME="<NODE_NAME>"
+NODE_IP="<NODE_IP>"
+  
+ETCDCTL_API=3 etcdctl snapshot restore ${WORKING_SNAPSHOT} \
+--data-dir /var/lib/etcd \
+--name ${NODE_NAME} \
+--initial-cluster ${NODE_NAME}=https://${NODE_IP}:2380 \
+--initial-advertise-peer-urls https://${NODE_IP}:2380
+```
+5. Start etcd on `master-0`.
++
+```bash
+mv /tmp/etcd.yaml /etc/kubernetes/manifests/
+```
+6. You should be able to see `master-0` joined to etcd cluster member list.
++
+```bash
+ETCDCTL_API=3 etcdctl \
+--endpoints=https://127.0.0.1:2379 \
+--cacert=/etc/kubernetes/pki/etcd/ca.crt 
+--cert=/etc/kubernetes/pki/etcd/healthcheck-client.crt \
+--key=/etc/kubernetes/pki/etcd/healthcheck-client.key member list
+```
+7. Add another master node to the etcd cluster member list.
++
+Replace <NODE_NAME> with cluster node name, for example: skuba-master-x
++
+Replace <NODE_IP> with cluster node IP address.
++
+HINT: the <NODE_NAME> and <NODE_IP> must match the correct node information of `--initial-cluster` in /etc/kubernetes/manifest/etcd.yaml of the targeting node.
+HINT: Nodes must be restored in sequence.
+`
++
+```bash
+NODE_NAME="<NODE_NAME>"
+NODE_IP="<NODE_IP>"
+
+ETCDCTL_API=3 etcdctl \
+--endpoints=https://127.0.0.1:2379 \
+--cacert=/etc/kubernetes/pki/etcd/ca.crt \
+--cert=/etc/kubernetes/pki/etcd/healthcheck-client.crt \
+--key=/etc/kubernetes/pki/etcd/healthcheck-client.key \
+member add ${NODE_NAME} --peer-urls=https://${NODE_IP}:2380
+```
+7. Remote to node on step 7.
+8. Start etcd.
++
+```
+cp /tmp/etcd.yaml /etc/kubernetes/manifests/
+```
+7. Repeat step 7, 8, 9 to recover all remain master nodes.
+
```

Comment:
Suggest to add item 10 to have a criteria to see what result means etcd recovery works and the cluster works correctly like K8s & etcd cluster healthy.